proposals/cli_schema.md

```diff
@@ -0,0 +1,307 @@
+
+1. The structure must be completely validated.
+2. It must allow backwards compatibility with existing test manifests out in the wild.
+   1. A test manifest which supplies the `apiVersion` and `kind` as specified in the proposed schema below is not subject to this backwards compatibility; only those in which these fields are absent should be subject to the existing logic.
+3. All file names must be checked for existence and validity.
+4. Objects in the `results[]` section should be validated against the rule type to determine if the necessary mandatory fields are present.
+5. Policies which are of type `Policy` must specify Namespace.
+6. There must not be duplicate names in resources.
+7. Deprecated fields must be handled with warnings to users.
+8. Breaking changes should cause the apiVersion to be incremented.
+
+```yaml
+apiVersion: cli.kyverno.io/v1beta1
+kind: KyvernoTest
+metadata:
+  name: test-check-nvidia-gpus
+  labels:
+    foolabel: foovalue
+  annotations:
+    fookey: foovalue
+spec:
+  policies:
+    # support fully-qualified file paths (https://github.com/kyverno/kyverno/issues/2315)
+    # wherever files are the value. Should be relative and absolute.
+  - check-nvidia-gpu.yaml
+  - ../relative/path/to/next/policy.yaml
+  - /fully/qualified/path/to/next/policy.yaml
+  resources:
+  - resource.yaml
+  - ../relative/path/to/next/resource.yaml
+  - /fully/qualified/path/to/next/resource.yaml
+  # a resource referenced with a dot as below means to check for another
+  # YAML document in this same test file. This could be changed so it's just an
+  # implicit check
+  - .
+    # a resource pool is an alternate concept which groups resources
+    # under a common name which can be accessed in the results[] section
+    # easier
+    my-resource-pool:
+    - resource.yaml
+    - ../relative/path/to/next/resource.yaml
+    - /fully/qualified/path/to/next/resource.yaml
+    - .
+  results:
+  # results[] vary depending on the rule type being tested.
+  # validate
+  - policy: policyname01 # required
+    rule: validaterulename # required
+    # resources must support multiple resources, all of which
+    # should be subject to the rest of the field values 
+    # in this result. (https://github.com/kyverno/kyverno/issues/2857)
+    resources: # required
+    - resource01
+    - resource02
+    - resource03
+    # resources should be specifiable with various naming formats like below
+    # GVK with namespace and resource name
+    - v1.Pod/nsfoo/mypod
+    # kind with namespace and resource name
+    - pod/nsfoo/mypod
+    # namespace and resource name
+    - nsfoo/mypod
+    # The below resources object is an alternative suggestion as how to reference
+    # resources based upon the resource pool defined above. It gives access to multiple
+    # resources in a flexible naming format and consistently so that providing a reference
+    # to a patched resource, for example, doesn't require specifying a file name.
+    resources:
```

Comment:
I like this in general as it gives more accessibility to test different states of objects in the same place.